app/invocation/invocation_artifacts_card.tsx

```diff
@@ -19,14 +19,14 @@ export default class ArtifactsCardComponent extends React.Component<Props, State
     numPages: 1,
   };
 
-  handleArtifactClicked(outputUri: string, outputFilename: string, event: MouseEvent) {
+  handleArtifactClicked(outputUri: string, outputFilename: string, event: React.MouseEvent<HTMLAnchorElement>) {
     event.preventDefault();
     if (!outputUri) return false;
 
     if (outputUri.startsWith("file://")) {
       window.prompt("Copy artifact path to clipboard: Cmd+C, Enter", outputUri);
     } else if (outputUri.startsWith("bytestream://")) {
-      rpcService.downloadBytestreamFile(outputFilename, outputUri, this.props.model.getId());
+      rpcService.downloadBytestreamFile(outputFilename, outputUri, this.props.model.getId() ?? "");
```

Comment:
should probably skip this if getId() is undefined no?

Reply:
So, yeah.  We have a lot of these now.  The production behavior of this today builds a URL containing "undefined" (i.e., this path is already unprotected).  Making interactions with invocation_model a bit more rigorous should probably be at the top of the list, but this doesn't make things worse and let's us get the bits flipped, so let's keep it separate for now.